experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py

```diff
@@ -232,75 +232,79 @@ def test_invalid_experiment_requires_min_version_less_than_max_version(self):
-            ).data,
-            context={"user": self.user},
-        )
-        self.assertTrue(serializer.is_valid())
-        self.assertEqual(
-            serializer.warnings,
-            {
-                "firefox_min_version": [NimbusExperiment.ERROR_DESKTOP_ROLLOUT_VERSION],
-            },
-        )
-
-    def test_desktop_min_version_over_113_no_warning(self):
-        experiment = NimbusExperimentFactory.create_with_lifecycle(
-            NimbusExperimentFactory.Lifecycles.CREATED,
-            application=NimbusExperiment.Application.DESKTOP,
-            firefox_min_version=NimbusExperiment.Version.FIREFOX_118,
-            is_rollout=True,
-            is_sticky=True,
-        )
-        for branch in experiment.treatment_branches:
-            branch.delete()
-        experiment.save()
-        serializer = NimbusReviewSerializer(
-            experiment,
-            data=NimbusReviewSerializer(
+    def test_rollout_min_version_under_115_shows_warning(self):
+        # Please give thoughts on the updates to this test I am thoroughly iffy about it
+        for application in [
+            NimbusExperiment.Application.DESKTOP,
+            NimbusExperiment.Application.FENIX,
+            NimbusExperiment.Application.FOCUS_ANDROID,
+            NimbusExperiment.Application.IOS,
+            NimbusExperiment.Application.FOCUS_IOS,
+        ]:
+            experiment = NimbusExperimentFactory.create_with_lifecycle(
+                NimbusExperimentFactory.Lifecycles.CREATED,
+                firefox_min_version=NimbusExperiment.Version.FIREFOX_114,
+                application=application,
+                is_rollout=True,
+                is_sticky=True,
+            )
+            for branch in experiment.treatment_branches:
+                branch.delete()
+            experiment.save()
+            serializer = NimbusReviewSerializer(
                 experiment,
+                data=NimbusReviewSerializer(
+                    experiment,
+                    context={"user": self.user},
+                ).data,
                 context={"user": self.user},
-            ).data,
-            context={"user": self.user},
-        )
-        self.assertTrue(serializer.is_valid())
-        self.assertEqual(serializer.warnings, {})
+            )
+            self.assertTrue(serializer.is_valid())
+            self.assertEqual(
+                serializer.warnings,
+                {
+                    "firefox_min_version": [
+                        # Ehhhhhh on this whole thing here? It's basically the same logic that exists in the actual code
+                        # I'm not a fan of it but idk how else to do it because I don't want to write the entire text out
+                        # Should I just write a separate test for each app?
+                        NimbusExperiment.ERROR_ROLLOUT_VERSION.format(
+                            application=NimbusExperiment.Application(application).label,
```

Comment:
yeahhh matching this stuff is a pita. This looks fine to me; maybe Yashika/Jared will have some better suggestions 😬